src/Lykke.Service.IroncladDecorator/Controllers/CallbackController.cs

```diff
@@ -67,24 +67,24 @@ public async Task<IActionResult> SigninCallback()
 
             var tokens = await GetTokens(authCode);
 
-            var userId = GetUserId(tokens.IdentityToken);
+            await SignInToLykke(tokens.IdentityToken, userSession, tokens);
 
-            var authResult = await _clientSessionsClient.Authenticate(userId, "hobbit");
+            var query = userSession.AuthorizeQuery;
 
-            SaveAuthResult(userSession, authResult);
+            var redirectUri = BuildFragmentRedirectUri(query, tokens);
 
-            SaveTokensToUserSession(userSession, tokens);
+            _log.Info("Redirecting to client app redirect uri. RedirectUri:{RedirectUri}", redirectUri);
+            return Redirect(redirectUri);
+        }
 
-            await SaveLykkeSession(authResult.SessionToken, tokens);
+        private async Task SignInToLykke(IdentityToken identityToken, UserSession userSession, TokenData tokens)
```

Comment:
Rename to `SignInToLykkeAsync`
```suggestion
        private async Task SignInToLykkeAsync(IdentityToken identityToken, UserSession userSession, TokenData tokens)
```